state/candidate.go

```diff
@@ -29,13 +30,18 @@ var (
 	ErrCandidateList = errors.New("invalid candidate list")
 )
 
-// Candidate indicates the structure of a candidate
-type Candidate struct {
-	Address       string
-	Votes         *big.Int
-	RewardAddress string
-	CanName       []byte // used as identifier to merge with native staking result, not part of protobuf
-}
+type (
+	// CandName is the 12-byte candidate name
+	CandName [12]byte
+
+	// Candidate indicates the structure of a candidate
+	Candidate struct {
+		Address       string
+		Votes         *big.Int
+		RewardAddress string
+		CanName       []byte // used as identifier to merge with native staking result, not part of protobuf
```

Comment:
Should we restrict the type or it has been too late with backward compatible problem?

Reply:
we could
but for safety (though the risk is low) would just leave it
we have func to12Bytes() to guarantee what we read from election must be 12-byte